### ISSUE TITLE
Repurpose README so it can be integrated into asciidoctor.org

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor-diagram#96]
The information about Asciidoctor Diagram on asciidoctor.org is not staying in sync with the project. What I'd like to do is have asciidoctor.org pull documentation from the project repository so it merely syndicates the information. However, I can't do that right now because there is information on the [Asciidoctor Diagram page](http://asciidoctor.org/docs/asciidoctor-diagram/) in asciidoctor.org that is not in the project's README.

To solve this problem, we need to think about how to repurpose the README so that it can be consumed by asciidoctor.org. There are two approaches:

* tag a section or sections in the README that should be used on asciidoctor.org
* write a dedicated intro and tutorial that asciidoctor.org can consume
* some combination of both

In short, I want the page on asciidoctor.org to consist only of include directives and a handful of attribute entries (such as `page-layout`). It should look something like:

```
:description: This guide describes the Asciidoctor Diagram extension for Asciidoctor.
:keywords: Asciidoctor diagram, AsciiDoc, Asciidoctor, PlantUML, sequence diagram,...
:page-layout: base
include::https://raw.githubusercontent.com/asciidoctor/asciidoctor-diagram/master/README.adoc[]
```

I think there is valuable information right now in the page on asciidoctor.org that we could pull into the README (or tutorial) in this project to help users get started with Asciidoctor Diagram. See https://github.com/asciidoctor/asciidoctor.org/blob/master/docs/asciidoctor-diagram.adoc

**Comment by pepijnve** [comment]
The page on asciidoctor.org is much friendlier and instructive compared to what I wrote myself.
What we could do is move the asciidoctor.org page to the asciidoctor-diagram page. Then we modify the asciidoctor.org page and the asciidoctor-diagram README to include that file.
That way I can keep things like the build status indicators, contributing info, etc in the README and we can share the tutorial style content.

**Comment by mojavelinux** [comment]
> What we could do is move the asciidoctor.org page to the asciidoctor-diagram page. Then we modify the asciidoctor.org page and the asciidoctor-diagram README to include that file.

That's exactly what I was thinking. I didn't want to just drop the content on asciidoctor.org because it provides a good starting point.

Would you like to attempt to merge the content into the README or do you need my assistance?

You are correct that includes still don't work on GitHub (and there's no indication of when they will). But that doesn't affect us in this case since we'll be pulling from GitHub.

**Comment by mojavelinux** [comment]
To clarify, you won't use any includes in the README file.

**Comment by pepijnve** [comment]
I'll give it a go first. So to summarise I'm going to

- Copy the content from asciidoctor.org to the asciidoctor-diagram repo as Usage.adoc
- Update Usage.adoc as needed so that it reflects the current state of the extension
- Remove the usage content from the asciidoctor-diagram README and replace that with a hyperlink to the new Usage page.

**Comment by mojavelinux** [comment]
Sounds like a good plan!

As you're making those changes, think about what parts we are going to include from the asciidoctor.org side. It sounds like we'll end up with two include directives. The first will be to pull the intro / description of the project. The second will be the detailed usage. Thus, the page on asciidoctor.org is merely a "view" of the content from the repository.

**Comment by pepijnve** [comment]
@mojavelinux wip is ready to review. Still need to finalise two of the advanced sections. The status part section is wrapped in an `ifndef::diagram_no_status[]` which should allow you to exclude that part from asciidoctor.org.

**Comment by mojavelinux** [comment]
I left some comments on the commit.

> Trying to figure out how I can give the TOC title a proper heading

You can specify custom text, including HTML:

```
:toc-title: pass:[:open_file_folder: <strong>Contents:</strong>]
```

Of course, you'll want to wrap that in an `ifdef::env-github` directive.

**Comment by mojavelinux** [comment]
Could you send a PR from your branch to make it easier to comment on the progress?

**Comment by mojavelinux** [comment]
Another approach for the toc is as follows:

```
= Asciidoctor Diagram
ifdef::env-github[:toc: macro:]

...

ifeval::["{toc-placement}" == "macro"]
[discrete]
== Contents

toc::[title={blank}]
endif::[]
```

That uses a regular section for the contents.

**Comment by pepijnve** [comment]
PR created. I'll integrate your comments on the commit already and fold those into the PR.

**Comment by mojavelinux** [comment]
:+1:  That answers the question I posted on the PR.

**Comment by pepijnve** [comment]
README has been revised in c5e2798621b76d0e678d725694304fabdb675509

**Comment by mojavelinux** [comment]
:+1: 

The README is now integrated into asciidoctor.org. For now, I'm linking to a specific revision until it is part of a release. Then we'll track the latest release using an attribute.

**Comment by mojavelinux** [comment]
See http://asciidoctor.org/docs/asciidoctor-diagram/

I had to make some small adjustments to the README, specifically to hide the toc in the site environment since we now generate it into the sidebar automatically.
